### PR TITLE
Add Trie State v1

### LINKED ISSUE #177: Add Trie version 1 logic
Image

More context: https://forum.polkadot.network/t/state-trie-migration/852

### PR DESCRIPTION
**Pull request by vikinatora**
This PR adds State Trie V1 logic. Specification can be found [here](https://spec.polkadot.network/chap-state#defn-node-header) and [here](https://spec.polkadot.network/chap-state#defn-hashed-subvalue). To summarize the changes
- Added encoding and decoding of hashed trie branch and leaf values in case the storage value is too long
- Added more tests to verify behaviour
- Refactored the code in order to improve code style and maintainability

Irrelevant to the state trie v1 changes:
- Added more westend bootnodes
- Added `@UtilityClass` to static classes

Closes #177

### CODE REVIEW

File: src/main/java/com/limechain/trie/decoder/TrieHeaderDecoder.java [hunk 1]
```diff
@@ -2,43 +2,58 @@
...
                 if (partialKeyLengthHeader > MAX_PARTIAL_KEY_LENGTH) {
                     throw new TrieDecoderException("Partial key overflow");
                 }
 
-                // Check if current byte is max byte value
+                // Check if current byte is less than max byte value
                 if (nextByte < 255) {
-                    return new TrieHeaderDecoderResult(variantBits, partialKeyLengthHeader, (byte) 0);
+                    return new TrieHeaderDecoderResult(header.nodeVariant(), partialKeyLengthHeader);
                 }
```
Comment (georg-getz) [inline src/main/java/com/limechain/trie/decoder/TrieHeaderDecoder.java:57]: Isn't is possible for the last byte to be exactly 255? Loop would continue to iterate. Is that expected behaviour? Maybe we can use `hasNext()` somewhere.

File: src/main/java/com/limechain/trie/decoder/TrieLeafDecoder.java [hunk 2]
```diff
@@ -1,26 +1,46 @@
...
+     * Decodes a leaf or hashed leaf node from a ScaleCodecReader input stream.
      *
-     * @param reader the ScaleCodecReader to read the encoded node data from
+     * @param reader           the ScaleCodecReader to read the encoded node data from
+     * @param variant          the variant of the node to be decoded
      * @param partialKeyLength the length of the partial key to be read
      * @return the decoded Node object
      * @throws TrieDecoderException if an error occurs while decoding the node. This could be
-     * due to an issue reading the children bitmap or the storage value.
+     *                              due to an issue reading the children bitmap or the storage value.
```
Comment (georg-getz) [inline src/main/java/com/limechain/trie/decoder/TrieLeafDecoder.java:22]: Is this indentation intentional?

Comment (vikinatora) [reply src/main/java/com/limechain/trie/decoder/TrieLeafDecoder.java:22]: Yes, it's meant to be aligned with the upper line
image

File: src/main/java/com/limechain/trie/decoder/TrieHeaderDecoder.java (hunk 1 above)
Comment (vikinatora) [reply src/main/java/com/limechain/trie/decoder/TrieHeaderDecoder.java:57]: I think that's fine and the code will execute as expected. 
image

Comment (vikinatora) [reply src/main/java/com/limechain/trie/decoder/TrieHeaderDecoder.java:57]: From here: https://spec.polkadot.network/chap-state#defn-node-header

**Review by georg-getz — APPROVED** [review]

**Pull request merged**